python/ccxt/base/decimal_to_precision.py

```diff
@@ -40,16 +40,16 @@ def decimal_to_precision(n, rounding_mode=ROUND, precision=None, counting_mode=D
     context.traps[decimal.Underflow] = True
     context.rounding = decimal.ROUND_HALF_UP  # rounds 0.5 away from zero
 
-    dec = decimal.Decimal(n)
-    string = str(dec)
+    dec = decimal.Decimal(str(n))
+    string = '{:f}'.format(dec)  # convert to string using .format to avoid engineering notation
     precise = None
 
     def power_of_10(x):
         return decimal.Decimal('10') ** (-x)
 
     if rounding_mode == ROUND:
         if counting_mode == DECIMAL_PLACES:
-            precise = str(dec.quantize(power_of_10(precision)))  # ROUND_HALF_EVEN is default context
+            precise = '{:f}'.format(dec.quantize(power_of_10(precision)))  # ROUND_HALF_EVEN is default context
```

Comment:
Did you test this with a number like 3456.00004?
I think that the quantize needs the number of digits of the whole number, not just after the floating point

Reply:
I did test it in my patch with the truncate mode, but did not in round node

Reply:
> Did you test this with a number like 3456.00004?

```
decimal_to_precision(3456.00004, TRUNCATE, 5, DECIMAL_PLACES)
Out[7]: '3456.00004'
decimal_to_precision(3456.00004, TRUNCATE, 4, DECIMAL_PLACES)
Out[8]: '3456'
```

We have done a lot of testing: https://github.com/ccxt/ccxt/blob/master/js/test/base/functions/test.number.js